[C-3294] Add Harmony PlainButton docs (#6678)
AudiusProject/audius-protocol: packages/harmony/src/components/button/PlainButton/PlainButton.mdx

import { Meta, Title, Description, Controls, Canvas, Primary, Stories, Unstyled } from '@storybook/blocks'

import { IconCaretDown, IconCloudUpload } from 'icons'
import { ComponentRules, Heading, BetaWarning } from 'storybook/components'

import { Box } from '../../layout'
import { PlainButtonSize } from '../types'
import { PlainButton } from './PlainButton'
import * as PlainButtonStories from './PlainButton.stories'

<Meta of={PlainButtonStories} />

<Title />

<BetaWarning />

- [Overview](#overview)
- [Props](#props)
- [Variants](#variants)
- [Configuration options](#configuration-options)
- [Use cases and examples](#use-cases-and-examples)
- [Do's and dont's](#dos-and-donts)

> - [Related Components](#related-components)

Reviewer: looks like there aren't any. Should add the section or remove the link

Author: Added 👍 . Good eye though

<Heading>
  ## Overview
  <Description />
</Heading>

<Canvas of={PlainButtonStories.Overview} />

## Props

<Controls />

## Variants

### Default

<Canvas of={PlainButtonStories.Default} />

### Subdued

<Canvas of={PlainButtonStories.Subdued} />

### Inverted

<Canvas of={PlainButtonStories.Inverted} />

## Configuration Options

### Sizes

<Canvas of={PlainButtonStories.Sizes} />

### Leading Icons

<Canvas of={PlainButtonStories.LeadingIcons} />

### Trailing Icons

<Canvas of={PlainButtonStories.TrailingIcons} />

## Use cases and examples

> - Plain buttons should be used as a tertiary button.

Reviewer: This is confusing as we have a Button variant called tertiary

Author: yep 😐

- Plain buttons are flexible and adaptable to any design. They can be emphasized or hidden.

> - Plain iconas are different from text link in there they may contain a leading or trailing icon.

Reviewer: ```suggestion
- Plain buttons are different from text links in that they may contain a leading or trailing icon.
```

There are also differences in their styling and the purpose of their use. Buttons can perform an arbitrary action, not just navigation.

## Do's and don'ts

<ComponentRules
  rules={[
    {
      positive: {
        description: 'Use plain buttons to reveal or hide additional information.',
        size: 'small',
        component: (

>           <Unstyled> 

Reviewer 2: Im thinking we add unstyled by default to the do's dont's just fyi

Author: I actually didnt need it here, but yea that might be good

            <PlainButton size={PlainButtonSize.LARGE} iconRight={IconCaretDown}>
              See More
            </PlainButton>
          </Unstyled>
        )
      },
      negative: {
        description: 'Don’t frame a plain button to create a “mock” secondary button.',
        size: 'small',
        component: (
          <Unstyled> 
            <Box pv='m' ph='xl' border='strong' borderRadius='s'>
              <PlainButton size={PlainButtonSize.LARGE} iconLeft={IconCloudUpload}>Upload</PlainButton>
            </Box>
          </Unstyled>
        )
      }
    }
  ]}
/>

> 

Reviewer 2: related components?

Author: Oh right, there weren't any in the figma docs. Button and SocialButton sound good here?
